src/AutoMapper/ConfigurationStore.cs

```diff
@@ -364,6 +364,17 @@ public TypeMap FindTypeMapFor(TypePair typePair)
             return typeMap;
         }
 
+        private bool CoveredByObjectMap(TypePair typePair)
+        {
+            return GetMappers().FirstOrDefault(m => m.IsMatch(new ResolutionContext(null, null, typePair.SourceType, typePair.DestinationType, new MappingOperationOptions(), new MappingEngine(this)))) != null;
+        }

```

Comment:
Cheap fix could be to null out the last 2 objects in this function call since they aren't being used.
Also I still agree with using TypePair instead of Resolution Context for IObjectMapper.IsMatch.